src/Analyzers/CSharp/Tests/UseCollectionInitializer/UseCollectionInitializerTests.cs

```diff
@@ -1826,4 +1827,42 @@ void M(List<int>? list1)
             LanguageVersion = LanguageVersion.CSharp12,
         }.RunAsync();
     }
+
+    [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/77416")]
+    public async Task TestNoCollectionExpressionForBlockingCollection()
+    {
+        await new VerifyCS.Test
+        {
+            TestCode = """
+                using System;
+                using System.Collections.Concurrent;
+
+                class A
+                {
+                    public void Main(ConcurrentQueue<int> queue)
+                    {
+                        BlockingCollection<int> bc = [|new|](queue);
+                        [|bc.Add(|]42);
+                    }
+                }
+                """,
+            FixedCode = """
+                using System;
+                using System.Collections.Concurrent;
+
+                class A
+                {
+                    public void Main(ConcurrentQueue<int> queue)
+                    {
+                        BlockingCollection<int> bc = new(queue)
```

Comment:
Eventually is `[with(queue), 42]` going to have the semantics we want here?

Reply:
yes, it would.